frontend/app/actions/RegistrationUri.scala

```diff
@@ -35,7 +36,12 @@ object RegistrationUri {
   )
 
   def parse(request: RequestHeader): String = {
```

Comment:
An observation - `parse()` is a really bad name for this method (I realise you didn't introduce this!) - half of the work it does is actually constructing the redirect url.